luigi/parameter.py

```diff
@@ -699,3 +706,87 @@ def parse(self, s):
 
     def serialize(self, e):
         return e.name
+
+
+class FrozenOrderedDict(collections.Mapping):
+    """
+    It is an immutable wrapper around ordered dictionaries that implements the complete :py:class:`collections.Mapping`
+    interface. It can be used as a drop-in replacement for dictionaries where immutability and ordering are desired.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.__dict = OrderedDict(*args, **kwargs)
+        self.__hash = None
+
+    def __getitem__(self, key):
+        return self.__dict[key]
+
+    def __iter__(self):
+        return iter(self.__dict)
+
+    def __len__(self):
+        return len(self.__dict)
+
+    def __repr__(self):
+        return '<FrozenOrderedDict %s>' % repr(self.__dict)
+
+    def __hash__(self):
+        if self.__hash is None:
+            hashes = map(hash, self.items())
+            self.__hash = functools.reduce(operator.xor, hashes, 0)
+
+        return self.__hash
+
+    def get_wrapped(self):
+        return self.__dict
+
+
+class DictParameter(Parameter):
+    """
+    Parameter whose value is a ``dict``.
+
+    In the task definition, use
+
+    .. code-block:: python
+
+        class MyTask(luigi.Task):
+          dict_param = luigi.DictParameter()
+
+            def run(self):
+                value = self.dict_param[key]
+
+    At the command line, use,
+
+    .. code-block:: console
+
+        $ luigi --module my_tasks MyTask --dict-param <JSON string>
```

Comment:
I think just saying `<JSON string>` is not clear for the average reader (at least not for me). Can you give an example too like `{"a": "1"}`, and also `{123: "456"}` if that were to be a valid value. You can keep this line but add another identical under it with `<JSON string>` replaced with an example value.